nautobot/core/graphql/types.py

```diff
@@ -34,3 +36,20 @@ class ContentTypeType(OptimizedNautobotObjectType):
 
     class Meta:
         model = ContentType
+
+
+class DateType(graphene.Date):
+    """
+    Overriding the default serialize method from https://github.com/graphql-python/graphene/blob/master/graphene/types/datetime.py
+    to handle the case where the date object is passed as a str object.
+    """
+
+    @staticmethod
+    def serialize(date):
+        if isinstance(date, datetime.datetime):
```

Comment:
If it's a datetime instance shouldn't we be serializing to a date and time not just the date?

Reply:
The DateType CustomField form selection widget only allows **date** and not **date and time**.  
![Screen Shot 2024-01-31 at 4 16 54 PM](https://github.com/nautobot/nautobot/assets/46973263/7da722df-539a-4aac-b8f0-d04c9901f33c). This is only an instance check and it should be represented as a `date` object in serialization IMO.

Reply:
It should be a datetime.date then, not a datetime.datetime. Probably an unrelated bug somewhere else in the code?

Reply:
Maybe the root cause of this issue though?

Reply:
This is a direct lift and shift from graphene https://github.com/graphql-python/graphene/blob/5fb7b543779360c8c3ffbf9d07e5769a14a1a0ed/graphene/types/datetime.py#L20. If you look at the definition of the `serialize` method here. It is checking if the date instance is a `datetime` type even tho it is within the `Date` class.

The root cause of the issue is that the date object we passed into the custom field is converted to a str type somewhere in the code (perhaps in the `_custom_field_data` conversion?) and that is what the elif statement fixed.

Reply:
Lets add the date check here to match the parent class. We'll need to update develop too ☹️

Reply:
Is this what you had in mind? https://github.com/nautobot/nautobot/pull/5204/commits/c25daac9832d06a3e456dbec32d24cc559234d34.
Or do you want me to take out the `if isinstance(date, datetime.datetime)` conditional entirely?